pkgs/development/python-modules/cocotb/default.nix

```
@@ -46,6 +47,14 @@ buildPythonPackage rec {
     substituteInPlace setup.py --replace "'cocotb-bus<1.0'" ""
   '';
 
+  # cocotb uses dlopen so that it's dynamic libraries are python version agnostic.
+  # Here we patch its dynamic libraries to make sure the correct libpython is found and used.
+  preFixup = ''
```

`.so` does not apply to darwin

```suggestion
  preFixup = lib.optionalString stdenv.hostPlatform.isLinux ''
```